packages/@mantine/core/src/components/NumberInput/NumberInput.tsx

```diff
@@ -358,6 +358,13 @@ export const NumberInput = factory<NumberInputFactory>((_props, ref) => {
     stepCountRef.current = 0;
   };
 
+  useEffect(() => {
+    if (inputRef.current) {
+      const { length } = inputRef.current.value;
+      inputRef.current.setSelectionRange(length, length);
```

Comment:
What if user wants to edit value in the middle, for example `251{cursor}213`, when anything is entered the selection range is moved to the end – this is unexpected behavior.

Reply:
I see. I've fixed this issue at [f49af28](https://github.com/mantinedev/mantine/pull/6004/commits/f49af28d6573c801a05be93e7b5dd138c8acd560). by this change, the position of cusor only changes when decrement/increment. I'll be happy if you check this change is Ok :)